include/exadg/structure/time_integration/driver_quasi_static_problems.cpp

```diff
@@ -226,17 +226,19 @@ DriverQuasiStatic<dim, Number>::solve_step(double const load_factor)
 
   output_solver_info_header(load_factor);
 
-  VectorType const const_vector;
-
   bool const update_preconditioner =
     this->param.update_preconditioner &&
     ((this->step_number - 1) % this->param.update_preconditioner_every_time_steps == 0);
 
-  auto const iter = pde_operator->solve_nonlinear(
+  VectorType const const_vector; // will not be used
+  auto const       iter = pde_operator->solve_nonlinear(
     solution, const_vector, 0.0 /*no mass term*/, load_factor /* = time */, update_preconditioner);
 
+  unsigned int const N_iter_nonlinear = std::get<0>(iter);
+  unsigned int const N_iter_linear    = std::get<1>(iter);
```

Comment:
```suggestion
  auto const    [N_iter_nonlinear, N_iter_linear]    = pde_operator->solve_nonlinear(
    solution, const_vector, 0.0 /*no mass term*/, load_factor /* = time */, update_preconditioner);
```

Reply:
Nice! I did this here for consistency with another occurrence in ExaDG. Hence, let's address this in a separate PR.

Reply:
Do we require C++17 (which is needed for this feature)?

Reply:
I think we already have C++17 in ExaDG (I think we needed it for filesystem related functionality). But as said before, let's discuss this elsewhere. I would suggest that @sebproell opens an issue with the "suggested changes" from above. Then, I can go through the code and replace several occurrences.